cmd/podman/login.go

```diff
@@ -60,19 +60,30 @@ func loginCmd(c *cli.Context) error {
 	if len(args) == 0 {
 		return errors.Errorf("registry must be given")
 	}
-	server := scrubServer(args[0])
+	server, isFullImageName := registryFromFullName(scrubServer(args[0]))
 	authfile := getAuthFile(c.String("authfile"))
 
 	sc := common.GetSystemContext("", authfile, false)
 
 	// username of user logged in to server (if one exists)
-	userFromAuthFile, err := config.GetUserLoggedIn(sc, server)
+	userFromAuthFile, passFromAuthFile, err := config.GetUserLoggedIn(sc, server)
 	if err != nil {
 		return errors.Wrapf(err, "error getting logged-in user")
 	}
-	username, password, err := getUserAndPass(c.String("username"), c.String("password"), userFromAuthFile)
-	if err != nil {
-		return errors.Wrapf(err, "error getting username and password")
+
+	var (
+		username string
+		password string
+	)
+	if isFullImageName && userFromAuthFile != "" {
+		username = userFromAuthFile
+		password = passFromAuthFile
+		fmt.Println("Authenticating with existing credentials...")
```

Comment:
Why does this case exist?

- If `userFromAuthFile`, the code below still does `SetAuthentication`, writing back exactly what has been just read. Is there some non-obvious reason why this is beneficial (reading from a legacy config file and writing to a new one, perhaps)? If so, there should be a comment here explaining this.
- It seems rather surprising to me that `podman login example.com` will always prompt for new credentials, but `podman login example.com/foo` will never do so. What is the connection between an ignored suffix and asking for a password?  [Might this be to be compatible with `docker login`? If so, again, please comment it so that someone does not came later and “fix” it.]
- If the credentials recorded in the config file are invalid (e.g. because the user has changed the password), in this case `podman login` will read the credentials from the file, try them, fail, report an error, and _give the user no opportunity to replace them_. That’s not useful, and the solution (drop the namespace suffix) is extremely non-obvious.

Reply:
Right, if we get a user and password from the authfile, we want to check that the credentials are still valid, hence we go through the authentication process and write back to the authfile if they are still valid.
The difference in the prompt is what docker login seems to do, I could change the case of `podman login example.com` to match that of `podman login example.com/foo`.
I actually didn't think of that case. Thanks for pointing it out, I will fix it to prompt for user and pass if the credentials are outdated.